storage/src/test/java/org/apache/kafka/storage/internals/log/OffsetMapTest.java

```diff
@@ -0,0 +1,121 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.nio.ByteBuffer;
+import java.security.DigestException;
+import java.security.NoSuchAlgorithmException;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class OffsetMapTest {
+
+    private static final int MEMORY_SIZE = 4096;
+
+    @ParameterizedTest
+    @ValueSource(ints = {10, 100, 1000, 5000})
+    public void testBasicValidation(int items) throws NoSuchAlgorithmException {
+        validateMap(items);
+    }
+
+    @Test
+    public void testClear() throws NoSuchAlgorithmException {
+        SkimpyOffsetMap map = new SkimpyOffsetMap(MEMORY_SIZE);
+        IntStream.range(0, 10).forEach(i -> assertDoesNotThrow(() -> map.put(key(i), i)));
+        IntStream.range(0, 10).forEach(i -> {
+            try {
+                assertEquals(i, map.get(key(i)));
+            } catch (DigestException e) {
+                throw new RuntimeException(e);
+            }
+        });
+        map.clear();
+        IntStream.range(0, 10).forEach(i -> {
+            try {
+                assertEquals(-1, map.get(key(i)));
+            } catch (DigestException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    @Test
+    public void testGetWhenFull() throws Exception {
+        SkimpyOffsetMap map = new SkimpyOffsetMap(MEMORY_SIZE);
+        int i = 37;
+        while (map.size() < map.slots()) {
+            map.put(key(i), i);
+            i = i + 1;
+        }
+        assertEquals(map.get(key(i)), -1);
+        assertEquals(map.get(key(i - 1)), i - 1);
+    }
+
+    @Test
+    public void testUpdateLatestOffset() throws Exception {
+        SkimpyOffsetMap map = new SkimpyOffsetMap(MEMORY_SIZE);
+        int i = 37;
+        while (map.size() < map.slots()) {
+            map.put(key(i), i);
+            i = i + 1;
+        }
+        int lastOffsets = 40;
+        assertEquals(map.get(key(i - 1)), i - 1);
+        map.updateLatestOffset(lastOffsets);
+        assertEquals(map.get(key(lastOffsets)), lastOffsets);
+    }
+
+    @Test
+    public void testLatestOffset() throws Exception {
+        SkimpyOffsetMap map = new SkimpyOffsetMap(MEMORY_SIZE);
+        int i = 37;
+        while (map.size() < map.slots()) {
+            map.put(key(i), i);
+            i = i + 1;
+        }
+        assertEquals(map.latestOffset(), i - 1);
+    }
+
+    private ByteBuffer key(Integer key) {
+        return ByteBuffer.wrap(key.toString().getBytes());
+    }
+
+    private void validateMap(int items) throws NoSuchAlgorithmException {
```

Comment:
we don't need this private method, right? It can be merged into `testBasicValidation`

Reply:
I will merge this method into `testBasicValidation`